### PR TITLE
Added Azure.SQLMI.AAD

### LINKED ISSUE #2117: Check that Azure AD authentication is configured for SQL Managed Instance
### Existing rule

_No response_

### Suggested rule

To SQL managed instance can have Azure AD authentication configured similar to SQL DB (`Azure.SQL.AAD`). Azure AD authentication should be used instead of SQL logins.

### Pillar

Security

### Additional context

- https://learn.microsoft.com/azure/azure-sql/database/authentication-azure-ad-only-authentication
- https://learn.microsoft.com/azure/azure-sql/database/authentication-aad-configure

Manged identity is required for this feature in SQL MI.

> You must enable the instance identity (SMI or UMI) to allow support for Azure AD authentication in SQL Managed Instance. For SQL Database, enabling the server identity is optional and required only if an Azure AD service principal (Azure AD application) oversees creating and managing Azure AD users, groups, or applications in the server. For more information, see [Azure Active Directory service principal with Azure SQL](https://learn.microsoft.com/azure/azure-sql/database/authentication-aad-service-principal?view=azuresql-mi).

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Fixes #2117 

Added Azure.SQLMI.AAD rule.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
@BernieWhite Ready for review :)

File: src/PSRule.Rules.Azure/rules/Azure.SQLMI.Rule.ps1 [hunk 1]
```diff
@@ -27,4 +27,29 @@ Rule 'Azure.SQLMI.AADOnly' -Ref 'AZR-000366' -Type 'Microsoft.Sql/managedInstanc
...
+                $Assert.HasFieldValue($config, 'properties.login')
+                $Assert.HasFieldValue($config, 'properties.sid')
+            }
+        }
+    }
+    else {
+        $Assert.HasFieldValue($TargetObject, 'properties.administratorType', 'ActiveDirectory')
+        $Assert.HasFieldValue($TargetObject, 'properties.login')
+        $Assert.HasFieldValue($TargetObject, 'properties.sid')
+    }
```
Comment (BernieWhite) [inline src/PSRule.Rules.Azure/rules/Azure.SQLMI.Rule.ps1:53]:
```suggestion
    }
}
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Looks good @BenjaminEngeset. One minor tweak required.

**Comment by BenjaminEngeset** [comment]
@BernieWhite fixed :)

**Review by BernieWhite — APPROVED** [review]
Great work @BenjaminEngeset! All good merge.

**Pull request merged**